crates/mqtt-format/src/v5/write.rs

```diff
@@ -15,17 +13,20 @@ pub enum MqttWriteError {
 pub trait WriteMqttPacket: Send {
     type Error: From<MqttWriteError>;
 
-    fn write_byte(&mut self, u: u8) -> WResult<Self>;
-    fn write_slice(&mut self, u: &[u8]) -> WResult<Self>;
+    fn write_byte(&mut self, u: u8) -> core::result::Result<(), <Self as WriteMqttPacket>::Error>;
+    fn write_slice(
+        &mut self,
+        u: &[u8],
+    ) -> core::result::Result<(), <Self as WriteMqttPacket>::Error>;
 
     #[inline]
-    fn write_u16(&mut self, u: u16) -> WResult<Self> {
+    fn write_u16(&mut self, u: u16) -> core::result::Result<(), <Self as WriteMqttPacket>::Error> {
         self.write_byte((u >> 8) as u8)?;
         self.write_byte(u as u8)
     }
 
     #[inline]
-    fn write_u32(&mut self, u: u32) -> WResult<Self> {
+    fn write_u32(&mut self, u: u32) -> core::result::Result<(), <Self as WriteMqttPacket>::Error> {
```

Comment:
Please use `Self::Error` in this case then

Reply:
Done